Sources/Optimizer/stochastic_reconfiguration.hpp

```diff
@@ -55,7 +55,7 @@ class SR {
         sr_diag_shift_(diagshift),
         use_iterative_(use_iterative),
         is_holomorphic_(is_holomorphic) {
-    InfoMessage() << GetInfoString();
+    InfoMessage() << LongDesc();
```

Comment:
Can we, please, not call `InfoMessage` here since it will be printed every time someone constructs an SR object in Python. If the user wants to see the description, they can call `info`.

Reply:
Agreed.

Maybe be should discuss this part a bit more:

What I did in thsi PR is to use `__repr__` to get a short representation of the object (useful, e.g., when testing things interactively in a shell or notebook).
In some cases, it can also be nice to have longer output describing an object (e.g., showing the Hamiltonian, Machine, SR, etc. for the driver). We should proably have a standard function to do that, which we can use for several classes. I would suggest calling it `.info()`, so that's what I did here.

What do you think?